### DISCUSSION TITLE
UGS Alarm on First G-Code Command

### DISCUSSION THREAD

**Discussion by Tapp1300** [winder/Universal-G-Code-Sender#2391] [Help]
Trying to run a Genmitsu ProverMAX 3020 with UGS. I have been through the Setup Wizard and everything checks out fine and works correctly.
I open a simple Sainsmart logo to carve. The spindle is in the home position, rear right corner. The UGS visualizer shows the spindle at the starting point of the carve, but the wood blank is at the front of the machine.
When I click on Run, I get an immediate error message on the first line. This happens no matter what project I try.
If I jog the spindle into the starting position for the carve, it doesn't even show up on the visualizer. 
I have attached photos of the 3020 in home position and what the Visualizer shows.
If the setup wizard says everything is fine, but it doesn't actually work, what am I missing?
![compressed_1702575274010](https://github.com/winder/Universal-G-Code-Sender/assets/153852585/e09040f9-a5af-44f8-80bc-44e7bfce1c38)
![ugs](https://github.com/winder/Universal-G-Code-Sender/assets/153852585/3db5e71b-a66c-4dc0-93d7-d76f14afd88e)

**Comment by breiler** [comment]
Did you set the work zero position?

You need to jog your machine to the closest left corner of your wood work piece and press the zero on all axes (press the X0, Y0, Z0 in the DRO window) to set your current work position.

**Comment by Tapp1300** [comment]
Joachim, 
Oh. My. Goodness. That was easy! At least it wasn't the E-stop! 
So I did that on another project, but when I hit Start I get a Hold message. 
It's a simple rectangular pocket tool path. Code goes like this:
G90G21(Pocket.To...)M05M0  ;T102M03S10000G0X-23.812Y-1.587Z31.750G1Z17.780F203.2Y1.588F508.0
I get the speed/position part, seems like there ought to be something in front of the semicolon in line 5.
Any thoughts? 
Thanks, J

SE! nt from AOL on Android 
 
  On Thu, Dec 14, 2023 at 12:47 PM, Joacim ***@***.***> wrote:   


Did you set the work zero position?

You need to jog your machine to the closest left corner of your wood work piece and press the zero on all axes (press the X0, Y0, Z0 in the DRO window) to set your current work position.
